src/session-adapters/cookieAdapter.ts

```diff
@@ -4,37 +4,48 @@ import { Adapter } from './generalAdapter'
 
 const clientSecret = process.env['CLIENT_SECRET'] || ''
 
+const createScopedKey = ({
+  spaceId,
+  userId,
+  key,
+}: {
+  spaceId: string
+  userId: string
+  key: string
+}) => {
+  return `${spaceId}:${userId}:${key}`
+}
+
+// We do not use `clientId` in cookie adapter,
+// because different plugins will have different domain names,
+// and it's enough to differentiate these cookie values.
 export const cookieAdapter: Adapter = {
-  getItem: ({ req, key }) => {
-    const cookie = getCookie(req, key)
+  getItem: ({ req, spaceId, userId, key }) => {
+    const cookie = getCookie(req, createScopedKey({ spaceId, userId, key }))
 
     if (!cookie) {
       return undefined
     }
 
-    //TODO: fix typing
-    return verifyData(clientSecret)(cookie) as object
+    const verifiedData = verifyData(clientSecret)(cookie)
+    if (!verifiedData) {
+      return undefined
+    } else {
+      return verifiedData as string
```

Comment:
`verifiedData` can never be a string as far as I understand from the method. I placed another comment suggesting having a generic to not cast.